vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,477 @@
+
+### Creating the `round` objects
+
+While the model tasks define what a model output submission should look like,
+the round defines the cadence of submissions and the timeframe under which a
+submission is considered valid.
+
+The most common way to create rounds is to create a single `round` object that
+tracks a date variable, in this case `"origin_date"`. The submission window for a given round is a period of one week, where the beginning of the window is
+4 days before the origin date and the end of the submission window is 2 days
+after the origin date. This means if you have an origin date of 
+`r format(as.Date("2023-01-02"), "%A %Y-%m-%d")`, then valid submission dates
+for that round are from 
+`r paste(format(as.Date("2023-01-02") + c(-4, 2), "%A %Y-%m-%d"), collapse = " to ")`
+
+```{r create-rounds}
+# Create a representation of a round as a list object of class `round`, the
+# `round1`
+round1 <- create_round(
+  round_id_from_variable = TRUE,
+  round_id = "origin_date",
+  round_name = "Round 1",
+  model_tasks = model_task_example,
+  submissions_due = list(
+    relative_to = "origin_date",
+    start = -4L,
+    end = 2L
+  )
+)
+
+# Combine the `round` objects to create a `rounds` class object, the `rounds`
+rounds <- create_rounds(round1)
+```
+
+### Creating and saving the `tasks` config file
+
+And the penultimate step is to create a config file and write it out to your
+hub.
+
+```{r create-tasks, eval = FALSE}
+# Create a representation of a complete `"tasks"` config file as a list object
+# of class `config`
+config <- create_config(rounds, output_type_id_datatype = "auto")
+
+# Write the `config` object to a JSON file (note that this will overwrite any
+# existing file with the same name in the "hub-config" folder)
+write_config(config, overwrite = TRUE)
+```
+```{r really-write}
+#| echo: false
+tmp <- tempfile(fileext = ".json")
+config <- create_config(rounds, output_type_id_datatype = "auto")
+write_config(config, config_path = tmp, overwrite = TRUE)
+```
+
+<details>
+<summary>contents of `hub-config/tasks.json`</summary>
+
+
+````json
+
+```{r show-config}
+#| echo: false
+#| results: asis
+writeLines(readLines(tmp))
+```
+
+````
+
+</details>
+
+
+## Validating your config file
+
+One of the strengths of specifying configuration in a JSON format is that it's
+machine-readable, which means that no matter how complex a JSON file gets, **it
+can be easily and rapidly validated**. All hubverse configuration files are
+validated against a central set of [hubverse
+schemas](https://github.com/hubverse-org/schemas/) that specify _how_ a
+configuration file is constructed. This allows any tool to be able to read a
+JSON file and validate it against the schema to ensure it's valid. 
+
+To check that your configuration file is valid, you can use the
+`validate_config()` function from the root directory of your hub:
+
+```{r validate-config-show}
+#| eval: false
+validate_config(config = "tasks")
```

Comment:
just a note that "tasks" is the default value of `config` so `validate_config()` would also work

Reply:
I included it specifically because during my first walkthrough of the hubverse documentation, having "tasks" as the default was not clear, so I wanted to highlight it here.